src/ColorScale.js

```diff
@@ -201,8 +223,17 @@ export default class ColorScale extends BaseClass {
             .map(d => quantile(allValues, d));
         }
         else {
-          const step = (domain[1] - domain[0]) / (colors.length - 1);
-          buckets = range(domain[0], domain[1] + step / 2, step);
+          if ((this._scale === "linear" || this._scale === "log") && diverging && this._color && this._centered) {
+            const negativeStep = ((this._midpoint - domain[0]) / Math.floor(colors.length / 2));
+            const positiveStep = ((domain[1] - this._midpoint) / Math.floor(colors.length / 2));
+            const negativeBuckets = range(domain[0], this._midpoint, negativeStep);
+            const positiveBuckets = range(this._midpoint, domain[1] + positiveStep / 2, positiveStep);
+
+            buckets = negativeBuckets.concat(positiveBuckets);
+          } else {
+            const step = (domain[1] - domain[0]) / (colors.length - 1);
+            buckets = range(domain[0], domain[1] + step / 2, step);
+          }
         }
```

Comment:
This whole code block is unecessarily nested (and contains a few eslint errors). Currently, it's this:

```js
else {
  if (...) {

  }
  else {

  }
}
```

And should be written like this:

```js
else if (...) {

}
else {

}
```